File: python/Level 1/숫자_문자열과_영단어.py
def solution(s):
    answer = ''
    i = 0
    nums = '0123456789'
    letters = ['zero', 'one', 'two', 'three', 'four',
               'five', 'six', 'seven', 'eight', 'nine']

    while i < len(s) :
        if s[i] in nums:
            answer += s[i]
            i += 1
        else:
            for n, letter in enumerate(letters):
                if s[i:].startswith(letter):
                    answer += str(n)
                    i += len(letter)
                    break
                
    return int(answer)

File: golang/Level 1/숫자_문자열과_영단어.go
import "strings"


func solution(s string) int {
    strToNum := make(map[string]int)
    strToNum["zero"] = 0
    strToNum["one"] = 1
    strToNum["two"] = 2
    strToNum["three"] = 3
    strToNum["four"] = 4
    strToNum["five"] = 5
    strToNum["six"] = 6
    strToNum["seven"] = 7
    strToNum["eight"] = 8
    strToNum["nine"] = 9
    strToNum["0"] = 0
    strToNum["1"] = 1
    strToNum["2"] = 2
    strToNum["3"] = 3
    strToNum["4"] = 4
    strToNum["5"] = 5
    strToNum["6"] = 6
    strToNum["7"] = 7
    strToNum["8"] = 8
    strToNum["9"] = 9
    
    answer := 0
    for i := 0; i < len(s); {
        for str, num := range strToNum {
            if strings.HasPrefix(s[i:], str) {
                answer *= 10
                answer += num
                i += len(str)
                break
            }
        }
    }

    return answer
}
